### PR TITLE
Fix allowed source prefix annotation validation

### PR DESCRIPTION
**Pull request by AloysAugustin**
## Description

This fixes the validator for the `AllowSpoofedSourcePrefixes` field in the `WorkloadEndpointSpec` object.

The validator was mistakenly using the `ip` tag while it should be `cidr`. It looks like the validation might not have been enforced in v3.23.0, but it is enforced in v3.23.1 and up, preventing pods with valid annotations from being created in recent versions of Calico.

## Related issues/PRs

None

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

Not required

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: libcalico-go/lib/apis/v3/workloadendpoint.go
Comment (fasaxc) [inline libcalico-go/lib/apis/v3/workloadendpoint.go:84]: How about adding a UT to make sure it doesn't get regressed again?

Comment (AloysAugustin) [reply libcalico-go/lib/apis/v3/workloadendpoint.go:84]: Good point, thanks @fasaxc , I have just added one.

**Comment by AloysAugustin** [comment]
@fasaxc Could you review this one again and add the `cherry-pick-candidate` label?

**Review by fasaxc — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
